express-notes-api/index.js

```diff
@@ -0,0 +1,92 @@
+const express = require('express');
+const app = express();
+const fs = require('fs');
+const json = require('./data.json');
+
+
+app.use(express.json())
+
+const arr = []
+//sets up json
+for (let key in json.notes) {
+  arr.push({ content: json.notes[key].content, id: key })
+}
+
+
+app.get('/api/notes', (req, res) => {
+
+  res.status(200).send(JSON.stringify(arr, null, 2))
+  return
+});
+
+app.get('/api/notes/:id', (req, res) => {
+
+  if (Number(req.params.id) < 0 || isNaN(req.params.id) )  { //checks for negatives and NaNs
+    res.status(400).json({ error: 'The id must be a positive integer, and no strings' });
+  }
+  else if (Number(req.params.id) >= json.nextId) {
+    res.json({ error: `cannot find note with id ${req.params.id}` })
+  }
+  else if (Number(req.params.id) === json.notes[req.params.id].id) {
```

Comment:
Your code would be easier to read if you just put `Number(req.params.id)` in a variable.